tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp

```diff
@@ -262,6 +262,12 @@ void parse_seconds() {
 
     fail_parse("1.2 1.3", "%S %S", time_ms);
     fail_parse("1.2 2.2", "%S %S", time_ms);
+
+    // LWG-3536, failed parse leaves duration unmodified.
+    minutes mm{20};
+    istringstream iss{"2:2:20"};
+    iss >> parse(string{"%H:%M:%S"}, mm);
+    assert(iss.fail() && mm == 20min);

```

Comment:
Is there any reason we couldn't use the helper function `fail_parse` here?

Reply:
`fail_parse` resets its argument.
https://github.com/microsoft/STL/blob/5ac862362900773fb95a8c9ab58fe49721e8cd92/tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp#L165-L168

It's more or less a copy-paste of `test_parse`, which I wrote that way because there are a few instances of parsing the same value different ways, and I wanted to be certain that I was actually asserting on a new value, not a stale one from a previous parse.
https://github.com/microsoft/STL/blob/5ac862362900773fb95a8c9ab58fe49721e8cd92/tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp#L243-L248